src/services/conversationService.ts

```diff
@@ -0,0 +1,379 @@
+import * as os from "os";
+import { CommentThread, ProgressLocation, Uri, window } from "vscode";
+import { extendErrorWithSupportLinks, provideSingleton } from "../utils";
+import { QueryManifestService } from "./queryManifestService";
+import { DBTProject } from "../manifest/dbtProject";
+import path = require("path");
+import { DBTTerminal } from "../dbt_client/dbtTerminal";
+import { AltimateRequest } from "../altimate";
+import { rmSync } from "fs";
+
+export interface SharedDoc {
+  share_id: number;
+  name: string;
+  description: string;
+  project_name: string;
+}
+
+export interface Conversation {
+  conversation_id: number;
+  message: string;
+  timestamp: string;
+  user_id: number;
+}
+
+export interface ConversationGroup {
+  conversation_group_id: number;
+  owner: number;
+  status: "Pending" | "Resolved";
+  meta: {
+    field?: "description";
+    column?: string;
+    highlight: string;
+    uniqueId?: string;
+    filePath: string;
+    resource_type?: string;
+    range: {
+      end: CommentThread["range"]["end"];
+      start: CommentThread["range"]["start"];
+    };
+  };
+  conversations: Conversation[];
+}
+
+@provideSingleton(ConversationService)
+export class ConversationService {
+  // Local cache to store shared docs
+  private sharedDocs: SharedDoc[] = [];
+  private conversationsBySharedDoc: Record<
+    SharedDoc["share_id"],
+    ConversationGroup[]
+  > = {};
+
+  public constructor(
+    private queryManifestService: QueryManifestService,
+    private dbtTerminal: DBTTerminal,
+    private altimateRequest: AltimateRequest,
+  ) {}
+
+  public getConversations() {
+    return this.conversationsBySharedDoc;
+  }
+
+  public async loadSharedDocs() {
+    try {
+      if (this.altimateRequest.getCredentialsMessage()) {
+        this.dbtTerminal.debug(
+          "ConversationService:loadSharedDocs",
+          "Missing credentials. skipping loadSharedDocs",
+        );
+        return;
+      }
+      // query the shared docs by current project names in workspace
+      const projectNames =
```

Comment:
Should we exclude duplicated projectNames?

Reply:
fixed